docs/docs/packages/cosmostxcollector.md

```diff
@@ -0,0 +1,192 @@
+This is the one used in the examples.
+
+### Example: Data collection
+
+The data collection example assumes that there is a PostgreSQL database running in the local
+environment containing an empty database named "cosmos".
+
+The required database tables will be created automatically by the collector the first time it is run.
+
+When the application is run it will fetch all the transactions and events starting from one of the
+recent blocks until the current block height and populate the database:
+
+```go
+package main
+
+import (
+	"context"
+	"log"
+
+	"github.com/ignite/cli/ignite/pkg/clictx"
+	"github.com/ignite/cli/ignite/pkg/cosmosclient"
+	"github.com/ignite/cli/ignite/pkg/cosmostxcollector"
+	"github.com/ignite/cli/ignite/pkg/cosmostxcollector/adapter/postgres"
+)
+
+const (
+	// Name of a local PostgreSQL database
+	dbName = "cosmos"
+
+	// Cosmos RPC address
+	rpcAddr = "https://rpc.cosmos.network:443"
+)
+
+func collect(ctx context.Context, db postgres.Adapter) error {
+	// Make sure that the data backend schema is up to date
+	if err = db.Init(ctx); err != nil {
+		return err
+	}
+
+	// Init the Cosmos client
+	client, err := cosmosclient.New(ctx, cosmosclient.WithNodeAddress(rpcAddr))
+	if err != nil {
+		return err
+	}
+
+	// Get the latest block height
+	latestHeight, err := client.LatestBlockHeight(ctx)
+	if err != nil {
+		return err
+	}
+
+	// Collect transactions and events starting from a block height.
+	// The collector stop at the latest height available at the time of the call.
+	collector := cosmostxcollector.New(db, client)
+	if err := collector.Collect(ctx, latestHeight-50); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func main() {
+	ctx := clictx.From(context.Background())
+
+	// Init an adapter for a local PostgreSQL database running with the default values
+	params := map[string]string{"sslmode": "disable"}
+	db, err := postgres.NewAdapter(dbName, postgres.WithParams(params))
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	if err := collect(ctx, db); err != nil {
+		log.Fatal(err)
+	}
+}
+```
+
+## Queries
+
+Collected data can be queried through the data backend adapters using event queries or
+cursor based queries.
+
+Queries support sorting, paging and filtering by using different options during creation.
+The cursor based ones also support the selection of specific fields or properties and also
+passing arguments in cases where the query is a function.
+
+### Event queries
+
+The event queries return events and their attributes as `[]cosmostxcollector.query.Event`.
+
+### Example: Query events
+
+```go
+import (
+	"context"
+
+	"github.com/ignite/cli/ignite/pkg/cosmostxcollector/adapter/postgres"
+	"github.com/ignite/cli/ignite/pkg/cosmostxcollector/query"
+)
+
+func queryEvents(ctx context.Context, db postgres.Adapter) ([]query.Event, error) {
+	// Create an events query that returns events of type "transfer"
+	qry := query.NewEventQuery(
+		query.WithFilters(
+			postgres.FilterByEventType("transfer"),
```

Comment:
Out of curiosity, is there a way to get the list of cosmos' events?

Reply:
Just added a paragraph to the docs that might clarify it. The default behaviour of a query is to select all, so selecting all events would be done by just creating the query as `qry := query.NewEventQuery()`.

Reply:
No problem, it's the a716a73

Reply:
Sorry again, my question wasn't clear enough. I meant to have a reference list of the different cosmos events. Here you have `transfer`, but what about the other ones from the legacy modules ?

Reply:
Assuming you don't want to read all events but just filter events for specific types, right now you would need to create a new filter similar to [FilterByEventTXs](https://github.com/ignite/cli/blob/develop/ignite/pkg/cosmostxcollector/adapter/postgres/filters.go#L120) by wrapping the `NewStringSliceFilter` or directly call it.

There are a couple of things I have in mind for improving the `cosmostxcollector` that I want to finish.

Reply:
I'm talking about event definitions like there https://github.com/cosmos/cosmos-sdk/blob/main/x/bank/types/events.go#L9 and there https://github.com/cosmos/cosmos-sdk/blob/main/x/gov/types/events.go#L5-L10.

What about adding that events definitions are usually stored in `x/mod/types/events.go` ? 

Also maybe replace `"transfer"` by `banktypes.EventTypeTransfer` ?

Reply:
Oh, I just actually understood what you meant in the initial comment!

Regarding the last comment, thats a really good addition to the docs 👍, it is a good practice to use the pre defined constants to filter by event type. I will add that then.